crates/matrix-sdk/src/event_cache/room/mod.rs

```diff
@@ -1140,6 +1152,20 @@ mod private {
             &self.events
         }
 
+        //// Find a single event in this room, starting from the most recent event.
+        ///
+        /// **Warning**! It looks into the loaded events in `RoomEvents`
+        /// **only**. It doesn't look inside the storage, contrary to
+        /// [`Self::find_event_by_id`].
+        pub fn rfind_event_in_memory_by<P>(&self, mut predicate: P) -> Option<Event>
+        where
+            P: FnMut(&Event) -> bool,
+        {
+            self.events
+                .revents()
+                .find_map(|(_position, event)| predicate(event).then_some(event.clone()))
```

Comment:
I think it's better to use `.then(|| Some(event.clone())` here, otherwise if the predicate is false, the cloned value will be unused (and thus the clone will be wasted).